File: app/src/main/java/com/mgaetan89/showsrage/activity/SettingsActivity.java
package com.mgaetan89.showsrage.activity;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.mgaetan89.showsrage.R;
import com.mgaetan89.showsrage.fragment.SettingsAboutFragment;
import com.mgaetan89.showsrage.fragment.SettingsBehaviorFragment;
import com.mgaetan89.showsrage.fragment.SettingsDisplayFragment;
import com.mgaetan89.showsrage.fragment.SettingsExperimentalFeaturesFragment;
import com.mgaetan89.showsrage.fragment.SettingsFragment;
import com.mgaetan89.showsrage.fragment.SettingsNetworkFragment;

public class SettingsActivity extends BaseActivity {
	@Override
	protected boolean displayHomeAsUp() {
		return true;
	}

	@Override
	protected Fragment getFragment() {
		return null;
	}

	@Override
	protected int getSelectedMenuId() {
		return R.id.menu_settings;
	}

	@Override
	protected int getTitleResourceId() {
		return R.string.settings;
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		if (savedInstanceState == null) {
			Uri data = this.getIntent().getData();
			String path = (data != null) ? data.getPath() : null;

			this.getFragmentManager().beginTransaction()
					.replace(R.id.content, getFragmentForPath(path))
					.commit();
		}
	}

	/* package */
	static SettingsFragment getFragmentForPath(@Nullable String path) {
		if (path != null) {
			switch (path) {
				case "/about":
					return new SettingsAboutFragment();

				case "/behavior":
					return new SettingsBehaviorFragment();

				case "/display":
					return new SettingsDisplayFragment();

				case "/experimental_features":
					return new SettingsExperimentalFeaturesFragment();

				case "/network":
					return new SettingsNetworkFragment();
			}
		}

		return new SettingsFragment();
	}
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/activity/SettingsActivity.kt
package com.mgaetan89.showsrage.activity

import android.os.Bundle
import com.mgaetan89.showsrage.R
import com.mgaetan89.showsrage.fragment.*

class SettingsActivity : BaseActivity() {
    override fun displayHomeAsUp() = true

    override fun getFragment() = null

    override fun getSelectedMenuId() = R.id.menu_settings

    override fun getTitleResourceId() = R.string.settings

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (savedInstanceState == null) {
            val path = this.intent.data?.path

            this.fragmentManager.beginTransaction()
                    .replace(R.id.content, getFragmentForPath(path))
                    .commit()
        }
    }

    companion object {
        fun getFragmentForPath(path: String?): SettingsFragment {
            return when (path) {
                "/about" -> SettingsAboutFragment()
                "/about/licenses" -> SettingsAboutLicensesFragment()
                "/about/showsrage" -> SettingsAboutShowsRageFragment()
                "/behavior" -> SettingsBehaviorFragment()
                "/display" -> SettingsDisplayFragment()
                "/experimental_features" -> SettingsExperimentalFeaturesFragment()
                "/network" -> SettingsNetworkFragment()
                "/network/api_key" -> SettingsNetworkApiKeyFragment()
                else -> SettingsFragment()
            }
        }
    }
}
